modules/search/instant-search/lib/api.js

```diff
@@ -124,9 +125,14 @@ function buildFilterObject( filterQuery ) {
 
 export function search( { aggregations, filter, pageHandle, query, resultFormat, siteId, sort } ) {
 	const key = stringify( Array.from( arguments ) );
-	const cachedVal = apiCache.get( key );
-	if ( cachedVal ) {
-		return cachedVal;
+
+	// Use cached value from the last 5 minutes
+	if ( cache.get( key ) ) {
+		return cache.get( key );
+	}
+	// Use cached value from the last 30 minutes if browser is offline
+	if ( ! navigator.onLine && backupCache.get( key ) ) {
```

Comment:
I was curious about browser support for this:

https://caniuse.com/#search=navigator.onLine

Looks really solid now 👍

Reply:
Yeah, I was quite surprised!